db/migrate/20150106050733_set_mysql_to_unicode_mb4.rb

```diff
@@ -25,24 +25,13 @@ def change_encoding(encoding, collation)
     execute "ALTER DATABASE `#{ActiveRecord::Base.connection.current_database}` CHARACTER SET #{encoding} COLLATE #{collation};"
 
     tables.each do |table|
-      execute "ALTER TABLE `#{table}` CHARACTER SET = #{encoding} COLLATE #{collation}"
-    end
 
-    character_columns.each do |table, columns|
-      columns.each do |column|
-        execute "ALTER TABLE `#{table}` CHANGE `#{column.name}` `#{column.name}` #{column.sql_type} CHARACTER SET #{encoding} COLLATE #{collation} #{column.null ? 'NULL' : 'NOT NULL'} #{"DEFAULT '#{column.default}'" if column.has_default?};"
-      end
-    end
-  end
+      text_columns = ""
+      columns(table).select {|column| column.type == :text }
+        .each {|column| text_columns += ", MODIFY `#{column.name}` TEXT #{'NOT' unless column.null } NULL #{"DEFAULT '#{column.default}'" if column.has_default?}" }
```

Comment:
I only insert it, when it is not empty, because the "," caused a SQL syntax error.